openhands/app_server/sandbox/remote_sandbox_service.py

```diff
@@ -303,21 +303,27 @@ async def get_sandbox_by_session_api_key(
         self, session_api_key: str
     ) -> Union[SandboxInfo, None]:
         """Get a single sandbox by session API key."""
-        # Get all stored sandboxes for the current user
-        stmt = await self._secure_select()
-        result = await self.db_session.execute(stmt)
-        stored_sandboxes = result.scalars().all()
-
-        # Check each sandbox's runtime data for matching session_api_key
-        for stored_sandbox in stored_sandboxes:
-            try:
-                runtime = await self._get_runtime(stored_sandbox.id)
-                if runtime and runtime.get('session_api_key') == session_api_key:
-                    return await self._to_sandbox_info(stored_sandbox, runtime)
-            except Exception:
-                # Continue checking other sandboxes if one fails
-                continue
-
+        try:
+            response = await self._send_runtime_api_request(
+                'GET',
+                '/list',
```

Comment:
Will this get all runtimes or only those that are running? (i.e is there an upper bound on the number of items?)